### PR TITLE
ensure initial route in the history

### PR DESCRIPTION
**Pull request by sertony**
This one is a possible fix for: https://www.notion.so/tonlabs/react-navigation-surf-is-changing-it-s-initial-screen-in-Split-View-on-tablets-2756114e6a024986870a9b570d5cd4a1

### CODE REVIEW

**Review by savelichalex — APPROVED** [review]

File: src/SurfSplitRouter.js [hunk 1]
```diff
@@ -268,6 +268,29 @@ export const SurfSplitRouter: RouterFactory<
...
+
+                // Ensure the history always includes the initial route
+                const initialRoute = state.routes.find(
+                    ({ name }) => name === initialRouteName,
+                );
+
+                if (initialRoute) {
+                    // Check if the history contains initial route already
+                    if (
+                        !newState.history ||
```
Comment (sertony) [inline src/SurfSplitRouter.js:280]: @savelichalex do you think we need to add the initial route into an empty history?
I honestly don't clearly understand in what case the history should be undefined...

File: src/SurfSplitRouter.js [hunk 2]
```diff
@@ -268,6 +268,29 @@ export const SurfSplitRouter: RouterFactory<
...
+                        !newState.history.find(
+                            // $FlowFixMe
+                            ({ key }) => key === initialRoute.key,
+                        )
+                    ) {
+                        // $FlowFixMe
+                        newState.history = [
```
Comment (savelichalex) [inline src/SurfSplitRouter.js:287]: what if we only have one screen and haven't moved to any screen yet, shouldn't we have an empty history?

File: src/SurfSplitRouter.js (hunk 1 above)
Comment (savelichalex) [reply src/SurfSplitRouter.js:280]: That's a good question actually, maybe it's just due to uncertainty of either we have history property in current state or not, when in reality it always should be...

**Comment by savelichalex** [comment]
Maybe it's time to cover it with tests?

File: src/SurfSplitRouter.js (hunk 2 above)
Comment (sertony) [reply src/SurfSplitRouter.js:287]: Looks like it's not empty and has an initial route itself:
https://cdn.discordapp.com/attachments/644127717553602561/824558780820750336/Screenshot_2021-03-25_at_11.22.06.png

**Review by savelichalex — APPROVED** [review]

**Pull request merged**